js/src/data/api-fetch-middlewares.test.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Internal dependencies
+ */
+import { createErrorResponseCatcher } from './api-fetch-middlewares';
+import { API_NAMESPACE } from './constants';
+
+describe( 'createErrorResponseCatcher', () => {
+	// Ref: https://github.com/WordPress/gutenberg/blob/%40wordpress/api-fetch%405.1.1/packages/api-fetch/src/index.js#L68-L81
+	const checkStatus = ( response ) => {
+		if ( response.status >= 200 && response.status < 300 ) {
+			return response;
+		}
+
+		throw response;
+	};
+
+	const createFetchHandler = ( status, jsonParsedData ) => () => {
+		const response = {
+			status,
+			json() {
+				return Promise.resolve( jsonParsedData );
+			},
+		};
+		return Promise.resolve( response ).then( checkStatus );
+	};
+
+	const passReject = ( response ) => Promise.reject( response );
+
+	const optionsDefaultParse = { path: `${ API_NAMESPACE }/hi` };
+	const optionsDontParse = { path: `${ API_NAMESPACE }/hi`, parse: false };
+
+	describe( 'force overwrite `parse` option to `false` and return the same result', () => {
+		let middleware;
+
+		beforeEach( () => {
+			middleware = createErrorResponseCatcher( passReject );
+		} );
+
+		it( 'should only handle requests where the path belongs to this plugin', async () => {
+			const data = {};
+			const optionsOtherNamespace = { path: '' };
+			const callback = jest.fn().mockImplementation( ( options ) => {
+				expect( options ).toBe( optionsOtherNamespace );
+				return data;
+			} );
+
+			const result = await middleware( optionsOtherNamespace, callback );
+			expect( callback ).toHaveBeenCalledTimes( 1 );
```

Comment:
:nail_care: ^ :nail_care: - like totally opinionated, codestyle costmetic.

Personally, I like to make a vertical space between action and assertion.

Reply:
Adjusted by https://github.com/woocommerce/google-listings-and-ads/pull/834/commits/1ef2778926b9298c0a3b4deee3f4ac7fae9ad1d6